flocker/node/agents/functional/test_ebs.py

```diff
@@ -68,6 +68,76 @@ class EBSBlockDeviceAPIInterfaceTests(
     """
     Interface adherence Tests for ``EBSBlockDeviceAPI``.
     """
+    def test_attach_when_foreign_device_has_next_device(self):
```

Comment:
This test seems far far too brittle and tied to implementation details. It would be better to just use public interfaces to recreate this:

1. Create volume with EC2 APIs directly.
2. Create volume with Flocker API, attempt to attach it, see that it fails. Or... doesn't fail, in which case diagnosis of problem is wrong or needs refining.